packages/amesos2/src/Amesos2_Umfpack.cpp

```diff
@@ -58,332 +57,14 @@ namespace Amesos2 {
 #endif
 #endif
 
-}
-
-
-#ifdef HAVE_TPETRA_INST_INT_INT
-namespace Amesos2 {
-#ifdef HAVE_TPETRA_INST_FLOAT
-  AMESOS2_SOLVER_TPETRA_INST(Umfpack,float,int,int);
-#endif
-#ifdef HAVE_TPETRA_INST_DOUBLE
-  AMESOS2_SOLVER_TPETRA_INST(Umfpack,double,int,int);
-#endif
-#ifdef HAVE_TPETRA_INST_COMPLEX_FLOAT
-  AMESOS2_SOLVER_TPETRA_INST(Umfpack,std::complex<float>,int,int);
-#endif
-#ifdef HAVE_TPETRA_INST_COMPLEX_DOUBLE
-  AMESOS2_SOLVER_TPETRA_INST(Umfpack,std::complex<double>,int,int);
-#endif
-}
-#endif //END INST_INT_INT
-
-#ifdef HAVE_TPETRA_INST_INT_UNSIGNED
-namespace Amesos2 {
-#ifdef HAVE_TPETRA_INST_FLOAT
-  AMESOS2_SOLVER_TPETRA_INST(Umfpack,float,int,unsigned int);
-#endif
-#ifdef HAVE_TPETRA_INST_DOUBLE
-  AMESOS2_SOLVER_TPETRA_INST(Umfpack,double,int,unsigned int);
-#endif
-#ifdef HAVE_TPETRA_INST_COMPLEX_FLOAT
-  AMESOS2_SOLVER_TPETRA_INST(Umfpack,std::complex<float>,int,unsigned int);
-#endif
-#ifdef HAVE_TPETRA_INST_COMPLEX_DOUBLE
-  AMESOS2_SOLVER_TPETRA_INST(Umfpack,std::complex<double>,int,unsigned int);
-#endif
-}
-#endif //END INST_INST_UNSIGNED
-
-
-#ifdef HAVE_TPETRA_INST_INT_LONG
-namespace Amesos2 {
-#ifdef HAVE_TPETRA_INST_FLOAT
-  AMESOS2_SOLVER_TPETRA_INST(Umfpack,float,int,long);
-#endif
-#ifdef HAVE_TPETRA_INST_DOUBLE
-  AMESOS2_SOLVER_TPETRA_INST(Umfpack,double,int,long);
-#endif
-#ifdef HAVE_TPETRA_INST_COMPLEX_FLOAT
-  AMESOS2_SOLVER_TPETRA_INST(Umfpack,std::complex<float>,int,long);
-#endif
-#ifdef HAVE_TPETRA_INST_COMPLEX_DOUBLE
-  AMESOS2_SOLVER_TPETRA_INST(Umfpack,std::complex<double>,int,long);
-#endif
-}
-#endif
-
-#ifdef HAVE_TPETRA_INST_INT_LONG_LONG
-namespace Amesos2 {
-#ifdef HAVE_TPETRA_INST_FLOAT
-  AMESOS2_SOLVER_TPETRA_INST(Umfpack,float,int,long long);
-#endif
-#ifdef HAVE_TPETRA_INST_DOUBLE
-  AMESOS2_SOLVER_TPETRA_INST(Umfpack,double,int,long long);
-#endif
-#ifdef HAVE_TPETRA_INST_COMPLEX_FLOAT
-  AMESOS2_SOLVER_TPETRA_INST(Umfpack,std::complex<float>,int,long long);
-#endif
-#ifdef HAVE_TPETRA_INST_COMPLEX_DOUBLE
-  AMESOS2_SOLVER_TPETRA_INST(Umfpack,std::complex<double>,int,long long);
-#endif
-}
-#endif
-//
-// mfh 16 Jan 2014: Hack to make explicit instantiation for
-// Ifpack2::Details::Amesos2Wrapper work.
-//
-#include "Kokkos_DefaultNode.hpp"
-#include "TpetraCore_ETIHelperMacros.h"
-
-#define AMESOS2_UMFPACK_LOCAL_INSTANT(S,LO,GO,N)                        \
-  template class Amesos2::Umfpack<Tpetra::CrsMatrix<S, LO, GO, N>,      \
-                                  Tpetra::MultiVector<S, LO, GO,  N> >;
+} //namespace Amesos2
 
+#define AMESOS2_UMFPACK_LOCAL_INSTANT(S,LO,GO,N) \
+  template class Amesos2::Umfpack<Tpetra::CrsMatrix<S, LO, GO, N>, \
```

Comment:
Note that in a complex arithmetic build, this requires that UMFPACK was built with complex arithmetic enabled.  @theguruat12 and @rrdrake know about this.

Reply:
@mhoemmen there is a TPL check to make sure the UMFPACK version supports complex types [see here](https://github.com/trilinos/Trilinos/blob/adbf724c2e749af6103daac1b041e578c3ce2525/cmake/TPLs/FindTPLUMFPACK.cmake#L57-L93), does this address your concern or do we need to add warning messages into Amesos2 as well?

Reply:
@ndellingwood I think that's OK -- the issue didn't come up before for Sierra because now they are switching to build Trilinos with CMake.  They should encounter this test now.  Thanks!